src/epoch/ops.rs

```diff
@@ -142,6 +145,105 @@ impl Epoch {
+    /// - target: targetted [TimeScale] we will transition to.
+    ///
+    /// Example:
+    /// ```
+    /// use hifitime::{Epoch, TimeScale, Polynomials, Unit};
+    ///
+    /// // random GPST Epoch for forward conversion to UTC
+    /// let t_gpst = Epoch::from_gregorian(2020, 01, 01, 0, 0, 0, 0, TimeScale::GPST);
+    ///
+    /// // Let's say we know the GPST-UTC polynomials for that day,
+    /// // They allow precise forward transition from GPST to UTC,
+    /// // and precise backward transition from UTC to GPST.
+    /// let gpst_utc_polynomials = Polynomials::from_constant_offset_nanoseconds(1.0);
+    ///
+    /// // This is the reference [Epoch] attached to the publication of these polynomials.
+    /// // You should use polynomials that remain valid and were provided recently (usually one day at most).
+    /// // Example: polynomials were published 1 hour ago.
+    /// let gpst_reference = t_gpst - 1.0 * Unit::Hour;
+    ///
+    /// // Forward conversion (to UTC) GPST - a0 + a1 *dt + a2*dt² = UTC
+    /// let t_utc = t_gpst.precise_timescale_conversion(true, gpst_reference, gpst_utc_polynomials, TimeScale::UTC)
+    ///     .unwrap();
+    ///
+    /// // Verify we did transition to UTC
+    /// assert_eq!(t_utc.time_scale, TimeScale::UTC);
+    ///
+    /// // Verify the resulting [Epoch] is the coarse GPST->UTC transition + fine correction
+    /// let reversed = t_utc.to_time_scale(TimeScale::GPST) + 1.0 * Unit::Nanosecond;
+    /// assert_eq!(reversed, t_gpst);
+    ///
+    /// // Apply the backward transition, from t_utc back to t_gpst.
+    /// // The timescale conversion works both ways: (from UTC) GPST = UTC + a0 + a1 *dt + a2*dt²
+    /// let backwards = t_utc.precise_timescale_conversion(false, gpst_reference, gpst_utc_polynomials, TimeScale::GPST)
+    ///     .unwrap();
+    ///
+    /// assert_eq!(backwards, t_gpst);
+    ///
+    /// // It is important to understand that your reference point does not have to be in the past.
+    /// // The only logic that should prevail is to always minimize interpolation gap.
+    /// // In other words, if you can access future interpolation information that would minimize the data gap, they should prevail.
+    /// // Example: +30' in the future.
+    /// let gpst_reference = t_gpst + 30.0 * Unit::Minute;
+    ///
+    /// // Forward conversion (to UTC) but using polynomials that were released 1 hour after t_gpst
+    /// let t_utc = t_gpst.precise_timescale_conversion(true, gpst_reference, gpst_utc_polynomials, TimeScale::UTC)
+    ///     .unwrap();
+    ///
+    /// // Verifications
+    /// assert_eq!(t_utc.time_scale, TimeScale::UTC);
+    ///
+    /// let reversed = t_utc.to_time_scale(TimeScale::GPST) + 1.0 * Unit::Nanosecond;
+    /// assert_eq!(reversed, t_gpst);
+    /// ```
+    pub fn precise_timescale_conversion(
+        &self,
+        forward: bool,
+        reference_epoch: Self,
+        polynomials: Polynomials,
```

Comment:
```suggestion
        polynomial: Polynomial,
```

There's a single polynomial here ;-)

Reply:
:) I'm not super good when it comes to grammar, and as years go by, everything seems to get worse (not only grammar 🤣 )